pkg/provider/provider.go

```diff
@@ -144,12 +144,18 @@ var (
 
 func buildTestEnvironment() { //nolint:funlen
 	start := time.Now()
+	env = TestEnvironment{}
 	// Wait for the debug pods to be ready before the autodiscovery starts.
+	oc := clientsholder.GetClientsHolder()
+	k8sPriviledgedDs.SetDaemonSetClient(oc.K8sClient)
 	err := k8sPriviledgedDs.WaitDaemonsetReady(DaemonSetNamespace, DaemonSetName, debugPodsTimeout)
 	if err != nil {
 		logrus.Errorf("Debug daemonset failure: %s", err)
+
+		// Because of this failure, we are only able to run a certain amount of tests that do not rely
+		// on the existence of the daemonset debug pods.
+		env.DaemonsetFailedToSpawn = true
```

Comment:
Is `env.DebugPods` be empty in case of the failure? It will skip the tests anyway.

Reply:
This also explicitly calls `ginkgo.Skip` so it'll show up as a proper skipped test rather than just a pass result.

Reply:
>It'll be partially populated
It is an option to zero all debug pods and remain with current skip mechanism.

>also explicitly calls
The failure prints error message anyway.

Overall, it is just a thought. I agree that more explicit handling - the better.

Reply:
Right but we don't want to fail some partner's workload because our debug pods fail to spawn causing confusion.